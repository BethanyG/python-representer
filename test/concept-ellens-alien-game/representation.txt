Module([ClassDef('placeholder_0', [], [], [Assign([Name('placeholder_1', Store(), lineno=22, col_offset=4, end_lineno=22, end_col_offset=24)], Constant(0, lineno=22, col_offset=27, end_lineno=22, end_col_offset=28), lineno=22, col_offset=4, end_lineno=22, end_col_offset=28), FunctionDef('placeholder_2', arguments([], [arg('self', lineno=24, col_offset=17, end_lineno=24, end_col_offset=21), arg('placeholder_3', lineno=24, col_offset=23, end_lineno=24, end_col_offset=35), arg('placeholder_4', lineno=24, col_offset=37, end_lineno=24, end_col_offset=49)], kwonlyargs=[], kw_defaults=[], defaults=[]), [AugAssign(Attribute(Name('placeholder_0', Load(), lineno=37, col_offset=8, end_lineno=37, end_col_offset=13), 'placeholder_1', Store(), lineno=37, col_offset=8, end_lineno=37, end_col_offset=34), Add(), Constant(1, lineno=37, col_offset=38, end_lineno=37, end_col_offset=39), lineno=37, col_offset=8, end_lineno=37, end_col_offset=39), Assign([Attribute(Name('self', Load(), lineno=39, col_offset=8, end_lineno=39, end_col_offset=12), 'placeholder_3', Store(), lineno=39, col_offset=8, end_lineno=39, end_col_offset=25)], Name('placeholder_3', Load(), lineno=39, col_offset=28, end_lineno=39, end_col_offset=40), lineno=39, col_offset=8, end_lineno=39, end_col_offset=40), Assign([Attribute(Name('self', Load(), lineno=40, col_offset=8, end_lineno=40, end_col_offset=12), 'placeholder_4', Store(), lineno=40, col_offset=8, end_lineno=40, end_col_offset=25)], Name('placeholder_4', Load(), lineno=40, col_offset=28, end_lineno=40, end_col_offset=40), lineno=40, col_offset=8, end_lineno=40, end_col_offset=40), Assign([Attribute(Name('self', Load(), lineno=41, col_offset=8, end_lineno=41, end_col_offset=12), 'placeholder_5', Store(), lineno=41, col_offset=8, end_lineno=41, end_col_offset=19)], Constant(3, lineno=41, col_offset=22, end_lineno=41, end_col_offset=23), lineno=41, col_offset=8, end_lineno=41, end_col_offset=23)], [], lineno=24, col_offset=4, end_lineno=41, end_col_offset=23), FunctionDef('placeholder_6', arguments([], [arg('self', lineno=43, col_offset=12, end_lineno=43, end_col_offset=16)], kwonlyargs=[], kw_defaults=[], defaults=[]), [AugAssign(Attribute(Name('self', Load(), lineno=52, col_offset=8, end_lineno=52, end_col_offset=12), 'placeholder_5', Store(), lineno=52, col_offset=8, end_lineno=52, end_col_offset=19), Sub(), Constant(1, lineno=52, col_offset=23, end_lineno=52, end_col_offset=24), lineno=52, col_offset=8, end_lineno=52, end_col_offset=24)], [], lineno=43, col_offset=4, end_lineno=52, end_col_offset=24), FunctionDef('placeholder_7', arguments([], [arg('self', lineno=54, col_offset=17, end_lineno=54, end_col_offset=21)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Return(Compare(Attribute(Name('self', Load(), lineno=60, col_offset=15, end_lineno=60, end_col_offset=19), 'placeholder_5', Load(), lineno=60, col_offset=15, end_lineno=60, end_col_offset=26), [Gt()], [Constant(0, lineno=60, col_offset=29, end_lineno=60, end_col_offset=30)], lineno=60, col_offset=15, end_lineno=60, end_col_offset=30), lineno=60, col_offset=8, end_lineno=60, end_col_offset=30)], [], lineno=54, col_offset=4, end_lineno=60, end_col_offset=30), FunctionDef('placeholder_8', arguments([], [arg('self', lineno=62, col_offset=17, end_lineno=62, end_col_offset=21), arg('placeholder_9', lineno=62, col_offset=23, end_lineno=62, end_col_offset=39), arg('placeholder_10', lineno=62, col_offset=41, end_lineno=62, end_col_offset=57)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Assign([Attribute(Name('self', Load(), lineno=70, col_offset=8, end_lineno=70, end_col_offset=12), 'placeholder_3', Store(), lineno=70, col_offset=8, end_lineno=70, end_col_offset=25)], Name('placeholder_9', Load(), lineno=70, col_offset=28, end_lineno=70, end_col_offset=44), lineno=70, col_offset=8, end_lineno=70, end_col_offset=44), Assign([Attribute(Name('self', Load(), lineno=71, col_offset=8, end_lineno=71, end_col_offset=12), 'placeholder_4', Store(), lineno=71, col_offset=8, end_lineno=71, end_col_offset=25)], Name('placeholder_10', Load(), lineno=71, col_offset=28, end_lineno=71, end_col_offset=44), lineno=71, col_offset=8, end_lineno=71, end_col_offset=44)], [], lineno=62, col_offset=4, end_lineno=71, end_col_offset=44), FunctionDef('placeholder_11', arguments([], [arg('self', lineno=73, col_offset=28, end_lineno=73, end_col_offset=32), arg('placeholder_12', lineno=73, col_offset=34, end_lineno=73, end_col_offset=39)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Pass(lineno=81, col_offset=8, end_lineno=81, end_col_offset=12)], [], lineno=73, col_offset=4, end_lineno=81, end_col_offset=12)], [], lineno=4, col_offset=0, end_lineno=81, end_col_offset=12), FunctionDef('placeholder_13', arguments([], [arg('placeholder_14', lineno=83, col_offset=26, end_lineno=83, end_col_offset=35)], kwonlyargs=[], kw_defaults=[], defaults=[]), [Return(ListComp(Call(Name('placeholder_0', Load(), lineno=90, col_offset=12, end_lineno=90, end_col_offset=17), [Subscript(Name('placeholder_15', Load(), lineno=90, col_offset=18, end_lineno=90, end_col_offset=26), Constant(0, lineno=90, col_offset=27, end_lineno=90, end_col_offset=28), Load(), lineno=90, col_offset=18, end_lineno=90, end_col_offset=29), Subscript(Name('placeholder_15', Load(), lineno=90, col_offset=31, end_lineno=90, end_col_offset=39), Constant(1, lineno=90, col_offset=40, end_lineno=90, end_col_offset=41), Load(), lineno=90, col_offset=31, end_lineno=90, end_col_offset=42)], [], lineno=90, col_offset=12, end_lineno=90, end_col_offset=43), [comprehension(Name('placeholder_15', Store(), lineno=90, col_offset=48, end_lineno=90, end_col_offset=56), Name('placeholder_14', Load(), lineno=90, col_offset=60, end_lineno=90, end_col_offset=69), [], 0)], lineno=90, col_offset=11, end_lineno=90, end_col_offset=70), lineno=90, col_offset=4, end_lineno=90, end_col_offset=70)], [], lineno=83, col_offset=0, end_lineno=90, end_col_offset=70)], [])