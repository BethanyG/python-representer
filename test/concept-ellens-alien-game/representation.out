# BEGIN TREE BEFORE
Module(
  [
    Expr(
      Constant(
        "Exemplar solution to Ellen's Alien Game exercise.",
        lineno=1,
        col_offset=0,
        end_lineno=1,
        end_col_offset=55),
      lineno=1,
      col_offset=0,
      end_lineno=1,
      end_col_offset=55),
    ClassDef(
      'Alien',
      [],
      [],
      [
        Expr(
          Constant(
            'Create an Alien object with location x_coordinate and y_coordinate.\n\n    Attributes\n    ----------\n    (class)total_aliens_created: int\n    x_coordinate: int - Position on the x-axis.\n    y_coordinate: int - Position on the y-axis.\n    health: int - Number of health points.\n\n    Methods\n    -------\n    hit(): Decrement Alien health by one point.\n    is_alive(): Return a boolean to indicate if Alien is alive (if health is > 0).\n    teleport(new_x_coordinate, new_y_coordinate): Move Alien object to new coordinates.\n    collision_detection(other): Implementation TBD.\n    ',
            lineno=5,
            col_offset=4,
            end_lineno=20,
            end_col_offset=7),
          lineno=5,
          col_offset=4,
          end_lineno=20,
          end_col_offset=7),
        Assign(
          [
            Name(
              'total_aliens_created',
              Store(),
              lineno=22,
              col_offset=4,
              end_lineno=22,
              end_col_offset=24)],
          Constant(
            0,
            lineno=22,
            col_offset=27,
            end_lineno=22,
            end_col_offset=28),
          lineno=22,
          col_offset=4,
          end_lineno=22,
          end_col_offset=28),
        FunctionDef(
          '__init__',
          arguments(
            [],
            [
              arg(
                'self',
                lineno=24,
                col_offset=17,
                end_lineno=24,
                end_col_offset=21),
              arg(
                'x_coordinate',
                lineno=24,
                col_offset=23,
                end_lineno=24,
                end_col_offset=35),
              arg(
                'y_coordinate',
                lineno=24,
                col_offset=37,
                end_lineno=24,
                end_col_offset=49)],
            kwonlyargs=[],
            kw_defaults=[],
            defaults=[]),
          [
            Expr(
              Constant(
                'Initialize a new Alien object and increment total_aliens_created by 1.\n\n        :param x_coordinate: int - Alien position on the x-axis\n        :param y_coordinate: int - Alien position on the y-axis\n\n        :attribute x_coordinate: int - Alien position on the x-axis\n        :attribute y_coordinate: int - Alien position on the y-axis\n        :attribute health: int (3) - Initial Alien health points.\n\n        :return: object - New Alien.\n        ',
                lineno=25,
                col_offset=8,
                end_lineno=35,
                end_col_offset=11),
              lineno=25,
              col_offset=8,
              end_lineno=35,
              end_col_offset=11),
            AugAssign(
              Attribute(
                Name(
                  'Alien',
                  Load(),
                  lineno=37,
                  col_offset=8,
                  end_lineno=37,
                  end_col_offset=13),
                'total_aliens_created',
                Store(),
                lineno=37,
                col_offset=8,
                end_lineno=37,
                end_col_offset=34),
              Add(),
              Constant(
                1,
                lineno=37,
                col_offset=38,
                end_lineno=37,
                end_col_offset=39),
              lineno=37,
              col_offset=8,
              end_lineno=37,
              end_col_offset=39),
            Assign(
              [
                Attribute(
                  Name(
                    'self',
                    Load(),
                    lineno=39,
                    col_offset=8,
                    end_lineno=39,
                    end_col_offset=12),
                  'x_coordinate',
                  Store(),
                  lineno=39,
                  col_offset=8,
                  end_lineno=39,
                  end_col_offset=25)],
              Name(
                'x_coordinate',
                Load(),
                lineno=39,
                col_offset=28,
                end_lineno=39,
                end_col_offset=40),
              lineno=39,
              col_offset=8,
              end_lineno=39,
              end_col_offset=40),
            Assign(
              [
                Attribute(
                  Name(
                    'self',
                    Load(),
                    lineno=40,
                    col_offset=8,
                    end_lineno=40,
                    end_col_offset=12),
                  'y_coordinate',
                  Store(),
                  lineno=40,
                  col_offset=8,
                  end_lineno=40,
                  end_col_offset=25)],
              Name(
                'y_coordinate',
                Load(),
                lineno=40,
                col_offset=28,
                end_lineno=40,
                end_col_offset=40),
              lineno=40,
              col_offset=8,
              end_lineno=40,
              end_col_offset=40),
            Assign(
              [
                Attribute(
                  Name(
                    'self',
                    Load(),
                    lineno=41,
                    col_offset=8,
                    end_lineno=41,
                    end_col_offset=12),
                  'health',
                  Store(),
                  lineno=41,
                  col_offset=8,
                  end_lineno=41,
                  end_col_offset=19)],
              Constant(
                3,
                lineno=41,
                col_offset=22,
                end_lineno=41,
                end_col_offset=23),
              lineno=41,
              col_offset=8,
              end_lineno=41,
              end_col_offset=23)],
          [],
          lineno=24,
          col_offset=4,
          end_lineno=41,
          end_col_offset=23),
        FunctionDef(
          'hit',
          arguments(
            [],
            [
              arg(
                'self',
                lineno=43,
                col_offset=12,
                end_lineno=43,
                end_col_offset=16)],
            kwonlyargs=[],
            kw_defaults=[],
            defaults=[]),
          [
            Expr(
              Constant(
                'Decrement Alien health by 1.\n\n        :return: None\n        ',
                lineno=44,
                col_offset=8,
                end_lineno=47,
                end_col_offset=11),
              lineno=44,
              col_offset=8,
              end_lineno=47,
              end_col_offset=11),
            AugAssign(
              Attribute(
                Name(
                  'self',
                  Load(),
                  lineno=52,
                  col_offset=8,
                  end_lineno=52,
                  end_col_offset=12),
                'health',
                Store(),
                lineno=52,
                col_offset=8,
                end_lineno=52,
                end_col_offset=19),
              Sub(),
              Constant(
                1,
                lineno=52,
                col_offset=23,
                end_lineno=52,
                end_col_offset=24),
              lineno=52,
              col_offset=8,
              end_lineno=52,
              end_col_offset=24)],
          [],
          lineno=43,
          col_offset=4,
          end_lineno=52,
          end_col_offset=24),
        FunctionDef(
          'is_alive',
          arguments(
            [],
            [
              arg(
                'self',
                lineno=54,
                col_offset=17,
                end_lineno=54,
                end_col_offset=21)],
            kwonlyargs=[],
            kw_defaults=[],
            defaults=[]),
          [
            Expr(
              Constant(
                'Return if the Alien is alive.\n\n        :return: boolean\n        ',
                lineno=55,
                col_offset=8,
                end_lineno=58,
                end_col_offset=11),
              lineno=55,
              col_offset=8,
              end_lineno=58,
              end_col_offset=11),
            Return(
              Compare(
                Attribute(
                  Name(
                    'self',
                    Load(),
                    lineno=60,
                    col_offset=15,
                    end_lineno=60,
                    end_col_offset=19),
                  'health',
                  Load(),
                  lineno=60,
                  col_offset=15,
                  end_lineno=60,
                  end_col_offset=26),
                [
                  Gt()],
                [
                  Constant(
                    0,
                    lineno=60,
                    col_offset=29,
                    end_lineno=60,
                    end_col_offset=30)],
                lineno=60,
                col_offset=15,
                end_lineno=60,
                end_col_offset=30),
              lineno=60,
              col_offset=8,
              end_lineno=60,
              end_col_offset=30)],
          [],
          lineno=54,
          col_offset=4,
          end_lineno=60,
          end_col_offset=30),
        FunctionDef(
          'teleport',
          arguments(
            [],
            [
              arg(
                'self',
                lineno=62,
                col_offset=17,
                end_lineno=62,
                end_col_offset=21),
              arg(
                'new_x_coordinate',
                lineno=62,
                col_offset=23,
                end_lineno=62,
                end_col_offset=39),
              arg(
                'new_y_coordinate',
                lineno=62,
                col_offset=41,
                end_lineno=62,
                end_col_offset=57)],
            kwonlyargs=[],
            kw_defaults=[],
            defaults=[]),
          [
            Expr(
              Constant(
                'Change Alien location.\n\n        :param new_x_coordinate: int - New location on x-axis.\n        :param new_y_coordinate: int - New location on y-axis.\n\n        :return: None\n        ',
                lineno=63,
                col_offset=8,
                end_lineno=69,
                end_col_offset=11),
              lineno=63,
              col_offset=8,
              end_lineno=69,
              end_col_offset=11),
            Assign(
              [
                Attribute(
                  Name(
                    'self',
                    Load(),
                    lineno=70,
                    col_offset=8,
                    end_lineno=70,
                    end_col_offset=12),
                  'x_coordinate',
                  Store(),
                  lineno=70,
                  col_offset=8,
                  end_lineno=70,
                  end_col_offset=25)],
              Name(
                'new_x_coordinate',
                Load(),
                lineno=70,
                col_offset=28,
                end_lineno=70,
                end_col_offset=44),
              lineno=70,
              col_offset=8,
              end_lineno=70,
              end_col_offset=44),
            Assign(
              [
                Attribute(
                  Name(
                    'self',
                    Load(),
                    lineno=71,
                    col_offset=8,
                    end_lineno=71,
                    end_col_offset=12),
                  'y_coordinate',
                  Store(),
                  lineno=71,
                  col_offset=8,
                  end_lineno=71,
                  end_col_offset=25)],
              Name(
                'new_y_coordinate',
                Load(),
                lineno=71,
                col_offset=28,
                end_lineno=71,
                end_col_offset=44),
              lineno=71,
              col_offset=8,
              end_lineno=71,
              end_col_offset=44)],
          [],
          lineno=62,
          col_offset=4,
          end_lineno=71,
          end_col_offset=44),
        FunctionDef(
          'collision_detection',
          arguments(
            [],
            [
              arg(
                'self',
                lineno=73,
                col_offset=28,
                end_lineno=73,
                end_col_offset=32),
              arg(
                'other',
                lineno=73,
                col_offset=34,
                end_lineno=73,
                end_col_offset=39)],
            kwonlyargs=[],
            kw_defaults=[],
            defaults=[]),
          [
            Expr(
              Constant(
                'Detect collisions with another Alien.\n\n        :param other: object - Other Alien object.\n\n        :return: None\n        ',
                lineno=74,
                col_offset=8,
                end_lineno=79,
                end_col_offset=11),
              lineno=74,
              col_offset=8,
              end_lineno=79,
              end_col_offset=11),
            Pass(
              lineno=81,
              col_offset=8,
              end_lineno=81,
              end_col_offset=12)],
          [],
          lineno=73,
          col_offset=4,
          end_lineno=81,
          end_col_offset=12)],
      [],
      lineno=4,
      col_offset=0,
      end_lineno=81,
      end_col_offset=12),
    FunctionDef(
      'new_aliens_collection',
      arguments(
        [],
        [
          arg(
            'positions',
            lineno=83,
            col_offset=26,
            end_lineno=83,
            end_col_offset=35)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Expr(
          Constant(
            'Create a list of Alien instances from a list of coordinate tuples.\n\n     :param positions: list - List of tuples of (x, y) coordinates.\n\n     :return: list - List of Alien objects.\n     ',
            lineno=84,
            col_offset=4,
            end_lineno=89,
            end_col_offset=8),
          lineno=84,
          col_offset=4,
          end_lineno=89,
          end_col_offset=8),
        Return(
          ListComp(
            Call(
              Name(
                'Alien',
                Load(),
                lineno=90,
                col_offset=12,
                end_lineno=90,
                end_col_offset=17),
              [
                Subscript(
                  Name(
                    'position',
                    Load(),
                    lineno=90,
                    col_offset=18,
                    end_lineno=90,
                    end_col_offset=26),
                  Constant(
                    0,
                    lineno=90,
                    col_offset=27,
                    end_lineno=90,
                    end_col_offset=28),
                  Load(),
                  lineno=90,
                  col_offset=18,
                  end_lineno=90,
                  end_col_offset=29),
                Subscript(
                  Name(
                    'position',
                    Load(),
                    lineno=90,
                    col_offset=31,
                    end_lineno=90,
                    end_col_offset=39),
                  Constant(
                    1,
                    lineno=90,
                    col_offset=40,
                    end_lineno=90,
                    end_col_offset=41),
                  Load(),
                  lineno=90,
                  col_offset=31,
                  end_lineno=90,
                  end_col_offset=42)],
              [],
              lineno=90,
              col_offset=12,
              end_lineno=90,
              end_col_offset=43),
            [
              comprehension(
                Name(
                  'position',
                  Store(),
                  lineno=90,
                  col_offset=48,
                  end_lineno=90,
                  end_col_offset=56),
                Name(
                  'positions',
                  Load(),
                  lineno=90,
                  col_offset=60,
                  end_lineno=90,
                  end_col_offset=69),
                [],
                0)],
            lineno=90,
            col_offset=11,
            end_lineno=90,
            end_col_offset=70),
          lineno=90,
          col_offset=4,
          end_lineno=90,
          end_col_offset=70)],
      [],
      lineno=83,
      col_offset=0,
      end_lineno=90,
      end_col_offset=70)],
  [])

# BEGIN TREE AFTER
Module(
  [
    ClassDef(
      'placeholder_0',
      [],
      [],
      [
        Assign(
          [
            Name(
              'placeholder_1',
              Store(),
              lineno=22,
              col_offset=4,
              end_lineno=22,
              end_col_offset=24)],
          Constant(
            0,
            lineno=22,
            col_offset=27,
            end_lineno=22,
            end_col_offset=28),
          lineno=22,
          col_offset=4,
          end_lineno=22,
          end_col_offset=28),
        FunctionDef(
          'placeholder_2',
          arguments(
            [],
            [
              arg(
                'self',
                lineno=24,
                col_offset=17,
                end_lineno=24,
                end_col_offset=21),
              arg(
                'placeholder_3',
                lineno=24,
                col_offset=23,
                end_lineno=24,
                end_col_offset=35),
              arg(
                'placeholder_4',
                lineno=24,
                col_offset=37,
                end_lineno=24,
                end_col_offset=49)],
            kwonlyargs=[],
            kw_defaults=[],
            defaults=[]),
          [
            AugAssign(
              Attribute(
                Name(
                  'placeholder_0',
                  Load(),
                  lineno=37,
                  col_offset=8,
                  end_lineno=37,
                  end_col_offset=13),
                'placeholder_1',
                Store(),
                lineno=37,
                col_offset=8,
                end_lineno=37,
                end_col_offset=34),
              Add(),
              Constant(
                1,
                lineno=37,
                col_offset=38,
                end_lineno=37,
                end_col_offset=39),
              lineno=37,
              col_offset=8,
              end_lineno=37,
              end_col_offset=39),
            Assign(
              [
                Attribute(
                  Name(
                    'self',
                    Load(),
                    lineno=39,
                    col_offset=8,
                    end_lineno=39,
                    end_col_offset=12),
                  'placeholder_3',
                  Store(),
                  lineno=39,
                  col_offset=8,
                  end_lineno=39,
                  end_col_offset=25)],
              Name(
                'placeholder_3',
                Load(),
                lineno=39,
                col_offset=28,
                end_lineno=39,
                end_col_offset=40),
              lineno=39,
              col_offset=8,
              end_lineno=39,
              end_col_offset=40),
            Assign(
              [
                Attribute(
                  Name(
                    'self',
                    Load(),
                    lineno=40,
                    col_offset=8,
                    end_lineno=40,
                    end_col_offset=12),
                  'placeholder_4',
                  Store(),
                  lineno=40,
                  col_offset=8,
                  end_lineno=40,
                  end_col_offset=25)],
              Name(
                'placeholder_4',
                Load(),
                lineno=40,
                col_offset=28,
                end_lineno=40,
                end_col_offset=40),
              lineno=40,
              col_offset=8,
              end_lineno=40,
              end_col_offset=40),
            Assign(
              [
                Attribute(
                  Name(
                    'self',
                    Load(),
                    lineno=41,
                    col_offset=8,
                    end_lineno=41,
                    end_col_offset=12),
                  'placeholder_5',
                  Store(),
                  lineno=41,
                  col_offset=8,
                  end_lineno=41,
                  end_col_offset=19)],
              Constant(
                3,
                lineno=41,
                col_offset=22,
                end_lineno=41,
                end_col_offset=23),
              lineno=41,
              col_offset=8,
              end_lineno=41,
              end_col_offset=23)],
          [],
          lineno=24,
          col_offset=4,
          end_lineno=41,
          end_col_offset=23),
        FunctionDef(
          'placeholder_6',
          arguments(
            [],
            [
              arg(
                'self',
                lineno=43,
                col_offset=12,
                end_lineno=43,
                end_col_offset=16)],
            kwonlyargs=[],
            kw_defaults=[],
            defaults=[]),
          [
            AugAssign(
              Attribute(
                Name(
                  'self',
                  Load(),
                  lineno=52,
                  col_offset=8,
                  end_lineno=52,
                  end_col_offset=12),
                'placeholder_5',
                Store(),
                lineno=52,
                col_offset=8,
                end_lineno=52,
                end_col_offset=19),
              Sub(),
              Constant(
                1,
                lineno=52,
                col_offset=23,
                end_lineno=52,
                end_col_offset=24),
              lineno=52,
              col_offset=8,
              end_lineno=52,
              end_col_offset=24)],
          [],
          lineno=43,
          col_offset=4,
          end_lineno=52,
          end_col_offset=24),
        FunctionDef(
          'placeholder_7',
          arguments(
            [],
            [
              arg(
                'self',
                lineno=54,
                col_offset=17,
                end_lineno=54,
                end_col_offset=21)],
            kwonlyargs=[],
            kw_defaults=[],
            defaults=[]),
          [
            Return(
              Compare(
                Attribute(
                  Name(
                    'self',
                    Load(),
                    lineno=60,
                    col_offset=15,
                    end_lineno=60,
                    end_col_offset=19),
                  'placeholder_5',
                  Load(),
                  lineno=60,
                  col_offset=15,
                  end_lineno=60,
                  end_col_offset=26),
                [
                  Gt()],
                [
                  Constant(
                    0,
                    lineno=60,
                    col_offset=29,
                    end_lineno=60,
                    end_col_offset=30)],
                lineno=60,
                col_offset=15,
                end_lineno=60,
                end_col_offset=30),
              lineno=60,
              col_offset=8,
              end_lineno=60,
              end_col_offset=30)],
          [],
          lineno=54,
          col_offset=4,
          end_lineno=60,
          end_col_offset=30),
        FunctionDef(
          'placeholder_8',
          arguments(
            [],
            [
              arg(
                'self',
                lineno=62,
                col_offset=17,
                end_lineno=62,
                end_col_offset=21),
              arg(
                'placeholder_9',
                lineno=62,
                col_offset=23,
                end_lineno=62,
                end_col_offset=39),
              arg(
                'placeholder_10',
                lineno=62,
                col_offset=41,
                end_lineno=62,
                end_col_offset=57)],
            kwonlyargs=[],
            kw_defaults=[],
            defaults=[]),
          [
            Assign(
              [
                Attribute(
                  Name(
                    'self',
                    Load(),
                    lineno=70,
                    col_offset=8,
                    end_lineno=70,
                    end_col_offset=12),
                  'placeholder_3',
                  Store(),
                  lineno=70,
                  col_offset=8,
                  end_lineno=70,
                  end_col_offset=25)],
              Name(
                'placeholder_9',
                Load(),
                lineno=70,
                col_offset=28,
                end_lineno=70,
                end_col_offset=44),
              lineno=70,
              col_offset=8,
              end_lineno=70,
              end_col_offset=44),
            Assign(
              [
                Attribute(
                  Name(
                    'self',
                    Load(),
                    lineno=71,
                    col_offset=8,
                    end_lineno=71,
                    end_col_offset=12),
                  'placeholder_4',
                  Store(),
                  lineno=71,
                  col_offset=8,
                  end_lineno=71,
                  end_col_offset=25)],
              Name(
                'placeholder_10',
                Load(),
                lineno=71,
                col_offset=28,
                end_lineno=71,
                end_col_offset=44),
              lineno=71,
              col_offset=8,
              end_lineno=71,
              end_col_offset=44)],
          [],
          lineno=62,
          col_offset=4,
          end_lineno=71,
          end_col_offset=44),
        FunctionDef(
          'placeholder_11',
          arguments(
            [],
            [
              arg(
                'self',
                lineno=73,
                col_offset=28,
                end_lineno=73,
                end_col_offset=32),
              arg(
                'placeholder_12',
                lineno=73,
                col_offset=34,
                end_lineno=73,
                end_col_offset=39)],
            kwonlyargs=[],
            kw_defaults=[],
            defaults=[]),
          [
            Pass(
              lineno=81,
              col_offset=8,
              end_lineno=81,
              end_col_offset=12)],
          [],
          lineno=73,
          col_offset=4,
          end_lineno=81,
          end_col_offset=12)],
      [],
      lineno=4,
      col_offset=0,
      end_lineno=81,
      end_col_offset=12),
    FunctionDef(
      'placeholder_13',
      arguments(
        [],
        [
          arg(
            'placeholder_14',
            lineno=83,
            col_offset=26,
            end_lineno=83,
            end_col_offset=35)],
        kwonlyargs=[],
        kw_defaults=[],
        defaults=[]),
      [
        Return(
          ListComp(
            Call(
              Name(
                'placeholder_0',
                Load(),
                lineno=90,
                col_offset=12,
                end_lineno=90,
                end_col_offset=17),
              [
                Subscript(
                  Name(
                    'placeholder_15',
                    Load(),
                    lineno=90,
                    col_offset=18,
                    end_lineno=90,
                    end_col_offset=26),
                  Constant(
                    0,
                    lineno=90,
                    col_offset=27,
                    end_lineno=90,
                    end_col_offset=28),
                  Load(),
                  lineno=90,
                  col_offset=18,
                  end_lineno=90,
                  end_col_offset=29),
                Subscript(
                  Name(
                    'placeholder_15',
                    Load(),
                    lineno=90,
                    col_offset=31,
                    end_lineno=90,
                    end_col_offset=39),
                  Constant(
                    1,
                    lineno=90,
                    col_offset=40,
                    end_lineno=90,
                    end_col_offset=41),
                  Load(),
                  lineno=90,
                  col_offset=31,
                  end_lineno=90,
                  end_col_offset=42)],
              [],
              lineno=90,
              col_offset=12,
              end_lineno=90,
              end_col_offset=43),
            [
              comprehension(
                Name(
                  'placeholder_15',
                  Store(),
                  lineno=90,
                  col_offset=48,
                  end_lineno=90,
                  end_col_offset=56),
                Name(
                  'placeholder_14',
                  Load(),
                  lineno=90,
                  col_offset=60,
                  end_lineno=90,
                  end_col_offset=69),
                [],
                0)],
            lineno=90,
            col_offset=11,
            end_lineno=90,
            end_col_offset=70),
          lineno=90,
          col_offset=4,
          end_lineno=90,
          end_col_offset=70)],
      [],
      lineno=83,
      col_offset=0,
      end_lineno=90,
      end_col_offset=70)],
  [])

# BEGIN NORMALIZED CODE AFTER
class placeholder_0:
    placeholder_1 = 0

    def placeholder_2(self, placeholder_3, placeholder_4):
        placeholder_0.placeholder_1 += 1
        self.placeholder_3 = placeholder_3
        self.placeholder_4 = placeholder_4
        self.placeholder_5 = 3

    def placeholder_6(self):
        self.placeholder_5 -= 1

    def placeholder_7(self):
        return self.placeholder_5 > 0

    def placeholder_8(self, placeholder_9, placeholder_10):
        self.placeholder_3 = placeholder_9
        self.placeholder_4 = placeholder_10

    def placeholder_11(self, placeholder_12):
        pass


def placeholder_13(placeholder_14):
    return [
        placeholder_0(placeholder_15[0], placeholder_15[1])
        for placeholder_15 in placeholder_14
    ]
